ui/components/app/srp-input/srp-input.test.js

```diff
@@ -0,0 +1,539 @@
+import React from 'react';
+import { waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import enLocale from '../../../../app/_locales/en/messages.json';
+import { renderWithLocalization } from '../../../../test/lib/render-helpers';
+import SrpInput from '.';
+
+const tooFewWords = new Array(11).fill('test').join(' ');
+const tooManyWords = new Array(25).fill('test').join(' ');
+const invalidWordCount = new Array(13).fill('test').join(' ');
+const invalidCorrectLength = new Array(12).fill('test').join(' ');
+const correct = `${new Array(11).fill('test').join(' ')} ball`;
+
+const invalidInputs = [
+  ' ',
+  'foo',
+  '🙂',
+  tooFewWords,
+  tooManyWords,
+  invalidWordCount,
+  invalidCorrectLength,
+];
+
+const poorlyFormattedInputs = [
+  ` ${correct}`,
+  `\t${correct}`,
+  `\n${correct}`,
+  `${correct} `,
+  `${correct}\t`,
+  `${correct}\n`,
+  `${new Array(11).fill('test').join('  ')}  ball`,
+  `${new Array(11).fill('test').join('\t')}\tball`,
+];
+
```

Comment:
>because it has the incorrect checksum

This is what the `invalidCorrectLength` test is for. That's an SRP that is invalid because it fails the checksum, but it is the correct length.

Reply:
We could add a test case with a word not on the wordlist :thinking: That's a good idea. I was also considering adding tests for the other locales we support for the SRP.

Reply:
Oh, I guess that is what invalidCorrectLength does

Reply:
Just added a test that includes a SRP with an invalid word, but a correct checksum.

I generated this word using this project: https://github.com/avsync/bip39chk
I altered the wordlist to include "aardvark", then I used that script to find a valid checksum word for the SRP with aardvark in it.

I tried adding a test for non-English locales as well, but apparently we only support English right now. The library we use to generate private key from the mnemonic in the background supports other locales, but the validation we're doing in the UI only works for English. So we can revisit that later if/when we expand it to support other locales.